replica/database.hh

```diff
@@ -554,14 +554,19 @@ private:
     compaction_group* single_compaction_group_if_available() const noexcept;
     // Select a compaction group from a given token.
     compaction_group& compaction_group_for_token(dht::token token) const noexcept;
+    // Returns the id of compaction group that owns token. Doesn't enforce that the group
+    // is actually allocated on this node.
+    size_t compaction_group_id_for_token(dht::token token) const noexcept;
```

Comment:
The patch deals with sparness on table initialization, but not on updates. After a while, tablets may be migrated away from the node. Do you have plans to handle that too?

Reply:
Yes, I am actually working on it right now, which is why I put the patch in draft mode.